src/openforms/validations/registry.py

```diff
@@ -50,7 +57,15 @@ def _flat(it):
     return list(_flat(iterables))
 
 
-class Registry(BaseRegistry):
+def filter_empty_values(value: Any) -> Any:
+    if not isinstance(value, dict):
+        # Let the serializer raise the validation error
+        return value
+
+    return {k: v for k, v in value.items() if v}
```

Comment:
Fixed it but still not happy with it. The thing is the frontend sends this kind of data:

```json
{
    "postCode": "...",
    "houseNumber": "...",
    "houseLetter": "",
    "houseNumberAddition": ""
}
```

That is it keeps empty strings for fields that were not filled by the user, thus failing with the serializer (I think because of the regex patterns) if not filtered

Reply:
well then your serializer needs to be updated to allow empty values :)

probably you have `required=True` set because that's the default and it's tripping up things.

Reply:
`required` is set correctly to `False`, what I need is something like _if provided but empty string, handle the field as being not provided, otherwise apply normal validation against the regex pattern_

Reply:
```py
>>> from rest_framework import serializers
>>> class Test(serializers.Serializer):
...     optional = serializers.RegexField(allow_blank=True, regex=r"\d+")
...     required = serializers.RegexField(allow_blank=False, regex=r"\d+")
...     
... 
>>> ser = Test(data={"optional": "", "required": "123"})
>>> ser.is_valid()
True
```